transcom/mymove
# [ MB-13693 ] Update SIT Display to show properly

## [Jira ticket](https://dp3.atlassian.net/browse/MB-13693)

## Summary

Prior to this, the SIT Display component would only update on a hard
page refresh instead of when the data became stale after the user
interacted with the service items on the shipment.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

It's difficult to explain how to test this PR since there are lot of different SIT scenarios that the SITDisplay and other components need to handle. First of all, there is the scenario that is called out in the Jira ticket, which can be tested with the following steps:

1. Create a move using the `HHGMoveIn200DaysSITNotApproved` test harness scenario.
2. Validate that the sit display is not shown when there are no approved sit service items
3. Approve the SIT service items and see the SIT Display updating on the page in
   real-time.

To cover everything else, it may be helpful to navigate through the [UI user flow](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?type=design&node-id=10285-112280&mode=design&t=JaQ0qzSmWRGalEww-4) and ensure that the different scenarios expressed in the user flow behave as expected. There are many testharness scenarios that are helpful here, including:

- HHGMoveIn200DaysSITDeparted
- HHGMoveIn200DaysSITEndsToday
- HHGMoveIn200DaysSITEndsTomorrow
- HHGMoveIn200DaysSITEndsYesterday
- HHGMoveIn200DaysSITNotApproved
- HHGMoveIn200DaysSITStartsInFuture
- HHGMoveIn200DaysSITWithPendingExtension
